lib/middleware/websocket/controller.rb

```diff
@@ -24,7 +24,10 @@ def on_shutdown(incoming_client)
     end
 
     def on_close(incoming_client)
-      puts 'websocket closed from the server'
+      closing_client =
+        @clients.select { |client| client.connection_client == incoming_client }
```

Comment:
Does this work in practice? Ie, can we guarantee that we'll always have the exact same object reference?

If that is indeed the case, I don't think we need to use `select`.

Reply:
confirmed this does work, gonna move forward since this method is likely to be changed in the future with unique identifiers. For now it does what it says it does, but has no real purpose.